README.md

```diff
@@ -1,12 +1,301 @@
 
 ## Description
 
-This is a kubernetes-native FRR wrapper to be consumed as a standalone component or together with MetalLB.
+The frr-k8s is deployed as a daemonset on Kubernetes clusters' nodes.
+
+Each FRR-K8s pod embeds an instance of [FRR](https://frrouting.org/).
+
+In order to configure each FRR, the various instances of the `FRRConfiguration` CRD are merged together and converted in
+a single configuration for the FRR container.
+
+<img src="docs/architecture.png" alt="architecture" width="600"/>
+
+## Installing
+
+The frr-k8s daemon provides 3 deployment methods.
+
+### All in one manifests
+
+This is the simples way to deploy and test frr-k8s, although the less configurable one.
+
+Just
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/metallb/frr-k8s/main/config/all-in-one/frr-k8s.yaml
+```
+
+Note: there is also a `frr-k8s-prometheus.yaml` variant which contains the opinionated configuration of metrics
+for prometheus deployed via the [kube-prometheus](https://github.com/prometheus-operator/kube-prometheus) repository.
+
+This is intended mostly for testing purpouses.
+
+### Kustomize
+
+You can install FRR-K8s with [Kustomize](https://kustomize.io/) by pointing at the remote kustomization file.
+
+```yaml
+# kustomization.yml
+namespace: frr-k8s-system
+
+resources:
+  - github.com/metallb/frr-k8s/config/frr-k8s?ref=main
+```
+
+### Helm
+
+You can install FRR-K8s with [Helm](https://helm.sh/) by using the Helm chart repository: `https://metallb.github.io/frr-k8s`:
+
+```bash
+helm repo add frr-k8s https://metallb.github.io/frr-k8s
+helm install frr-k8s frr-k8s/frr-k8s
+```
+
+A values file may be specified on installation. This is recommended for providing configs in Helm values:
+
+```bash
+helm install frr-k8s frr-k8s/frr-k8s -f values.yaml
+```
+
+**NOTE:** The frr-k8s pod requires elevated permission in order to perform its network functionalities.
+
+If you are using frr-k8s with a kubernetes version that enforces pod security admission (which is beta in k8s 1.23), the namespace frr-k8s is deployed to must be labelled with:
+
+```yaml
+  labels:
+    pod-security.kubernetes.io/enforce: privileged
+    pod-security.kubernetes.io/audit: privileged
+    pod-security.kubernetes.io/warn: privileged
+```
+
+## Configuring FRR-K8s
+
+The CRD to configure `FRR-K8s` is the `FRRConfiguration`. A comprehensive documentation of the API can be found in
+the [API Docs file](./API-DOCS.md)
+
+### BGP Configuration
+
+The `BGP` section of the FRRConfiguration spec is in charge of configuring the details related to the BGP protocol.
+
+#### Configuring one (or multiple) routers
+
+It is possible to configure multiple routers, one for each VRF.
+
+For each router, you can define the ASN and the VRF it belongs to, plus a list of prefixes to announce
+and a list of BGP neighbors to connect to:
+
+```yaml
+spec:
+  bgp:
+    routers:
+    - asn: 64512
+      neighbors:
```

Comment:
maybe also add a `toAdvertise` since we declare prefixes?

Reply:
toAdvertise is expanded later